calicoctl/calico_ctl/pool.py

```diff
@@ -122,19 +122,58 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
-    for cidr in cidrs:
+    current_pools = client.get_ip_pools(version)
+    current_pool_nets = [IPNetwork(pool.cidr) for pool in current_pools]
+
+    new_pools = []
+
+    for new_cidr in cidrs:
+        new_cidr_net = IPNetwork(new_cidr)
+
+        # Check if new pool overlaps with any existing pool
+        overlapping_old_cidr = _check_if_cidr_overlaps_cidrs(new_cidr,
+                                   [net.cidr for net in current_pool_nets])
+        if overlapping_old_cidr:
+            print "Cannot add IP pool %s - pool conflicts with exisitng " \
+                  "pool %s" % (new_cidr, overlapping_old_cidr)
+            sys.exit(1)
+
+        # Check if new pool overlaps with any other new pool
+        overlapping_new_cidr = _check_if_cidr_overlaps_cidrs(new_cidr,
+                                   [pool.cidr for pool in new_pools])
+        if overlapping_new_cidr:
+            print "Cannot add IP pool %s - pool conflicts with another " \
+                  "new pool %s" % (new_cidr, overlapping_new_cidr)
+            sys.exit(1)
+
+        # Add cidr as a new IP pool
```

Comment:
Can you modify this comment to indicate that they are not actually being added, but instead: `# Add cidr to list of pools to be added`

My original pass through I thought they were being added here, which would be bad as the script would exit out halfway through. It's good that that isn't the case, but bad that the comment makes it seem so